zingocli/tests/integration_tests.rs

```diff
@@ -9,13 +9,272 @@ use json::JsonValue;
+    //     4.2. check that the wallet is empty
+    //     4.3. rescan
+    //     4.4. check that notes and utxos were detected by the wallet
+    //
+    // # Current watch-only mode limitations:
+    // - wallet will not detect funds on all transparent addresses
+    //   see: https://github.com/zingolabs/zingolib/issues/245
+    // - wallet will not detect funds on internal addresses
+    //   see: https://github.com/zingolabs/zingolib/issues/246
+
+    let (regtest_manager, child_process_handler, mut client_builder) = scenarios::custom_clients();
+    let faucet = client_builder.build_new_faucet(0, false);
+    let original_recipient =
+        client_builder.build_newseed_client(HOSPITAL_MUSEUM_SEED.to_string(), 0, false);
+    let (zingo_config, _) = client_builder.make_new_zing_configdir();
+
+    Runtime::new().unwrap().block_on(async {
+        let (recipient_taddr, recipient_sapling, recipient_unified) = (
+            get_base_address!(original_recipient, "transparent"),
+            get_base_address!(original_recipient, "sapling"),
+            get_base_address!(original_recipient, "unified"),
+        );
+        let addr_amount_memos = vec![
+            (recipient_taddr.as_str(), 1_000u64, None),
+            (recipient_sapling.as_str(), 2_000u64, None),
+            (recipient_unified.as_str(), 3_000u64, None),
+        ];
+        // 1. fill wallet with a coinbase transaction by syncing faucet with 1-block increase
+        utils::increase_height_and_sync_client(&regtest_manager, &faucet, 1).await;
+        // 2. send a transaction contaning all types of outputs
+        faucet.do_send(addr_amount_memos).await.unwrap();
+        utils::increase_height_and_sync_client(&regtest_manager, &original_recipient, 1).await;
+        let sent_t_value = original_recipient.do_balance().await["transparent_balance"].clone();
+        let sent_s_value = original_recipient.do_balance().await["sapling_balance"].clone();
+        let sent_o_value = original_recipient.do_balance().await["orchard_balance"].clone();
+        assert_eq!(sent_t_value, 1000u64);
+        assert_eq!(sent_s_value, 2000u64);
+        assert_eq!(sent_o_value, 3000u64);
+
+        // check that do_rescan works
+        original_recipient.do_rescan().await.unwrap();
+        assert_eq!(
+            original_recipient.do_balance().await["transparent_balance"],
+            1000u64
+        );
+        assert_eq!(
+            original_recipient.do_balance().await["sapling_balance"],
+            2000u64
+        );
+        assert_eq!(
+            original_recipient.do_balance().await["orchard_balance"],
+            3000u64
+        );
+
+        // Extract viewing keys
+        let wc = original_recipient
+            .extract_unified_capability()
+            .read()
+            .await
+            .clone();
+        use zcash_primitives::sapling::keys::DiversifiableFullViewingKey as SaplingFvk;
+        //let extfvk: SaplingFvk = (&wc).try_into().unwrap();
```

Comment:
What's the deal with this?   It seems to be a different version of:

`let s_fvk = Fvk::Sapling(SaplingFvk::try_from(&wc).unwrap().to_bytes());`

a few lines later.  I'd like a reviewer to sanity check me on that before I remove the line.

Reply:
I removed it.